rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/RascalLanguageServer.java

```diff
@@ -140,7 +161,84 @@ public WorkspaceService getWorkspaceService() {
 
         @Override
         public void connect(LanguageClient client) {
+            this.client = (IRascalLanguageClient) client;
+            this.ideServicesConfiguration = startIDEServices(this.client);
             getTextDocumentService().connect(client);
         }
+
+        /**
+         * Start a server for remote IDE services. These are used
+         * by an implementation of @see IDEServices that is given to Rascal terminal REPLS
+         * when they are started in the context of the LSP. This way
+         * Rascal REPLs get access to @see IDEServices to provide browsers 
+         * for interactive visualizations, starting editors and resolving IDE project URI.
+         * 
+         * @return the port number that the IDE services are running on
+         * @throws IOException when a new server socket can not be established.
+         */
+        private IDEServicesConfiguration startIDEServices(IRascalLanguageClient client) {
+            try {
+                ServerSocket socket = new ServerSocket(0);
+            
+                new IDEServicesThread(client, socket).start();
+            
+                return new IDEServicesConfiguration(socket.getLocalPort());
+            } catch (IOException e) {
+                logger.error(e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        private static class IDEServicesThread extends Thread {
+            private final IRascalLanguageClient ideClient;
+            private final ServerSocket serverSocket;
+
+            public IDEServicesThread(IRascalLanguageClient client, ServerSocket socket) {
+                super("Terminal IDE Services Thread");
+                setDaemon(true);
+                this.serverSocket = socket;
+                this.ideClient = client;
+            }
+
+            @Override
+            public void run() {
+                try {
+                    while(true) {
+                        logger.trace("Accepting connection for TerminalIDE Services on port {}", serverSocket.getLocalPort());
```

Comment:
I think this message should only be printed once (so outside the while loop), now it prints it after every successful connect).

Reply:
I needed that print but it can go now.

Reply:
I moved it out of the while loop. still nice to see the server is ready.